File: Back/NodeJS/tests/unit/implems/StubAddFriendPresenter.ts
import { IAddFriendOutputPort } from "../../../src/application/features/add-friend/AddFriendHandler";

export class StubAddFriendPresenter implements IAddFriendOutputPort {
    MemberDoesNotExist(MemberId: string): void {
        this.FriendAdded = false;
    }
    Present(): void {
        this.FriendAdded = true;
    }
    FriendAdded?: boolean;
}

File: Back/.Net/Modules/Bets/BetFriends.Bets.Application.UnitTests/Implems/StubAddFriendPresenter.cs
using BetFriends.Bets.Application.Features.AddFriend;

namespace BetFriends.Bets.Application.UnitTests.Implems;

internal class StubAddFriendPresenter : IAddFriendOutputPort
{
    public StubAddFriendPresenter()
    {
    }

    public bool? FriendAdded { get; private set; }

    public void MemberDoesNotExist()
    {
        FriendAdded = false;
    }

    public void Present()
    {
        FriendAdded = true;
    }
}
